addon/serializers/json-api.js

```diff
@@ -178,7 +179,11 @@ const JSONAPISerializer = JSONSerializer.extend({
   */
   pushPayload(store, payload) {
     let normalizedPayload = this._normalizeDocumentHelper(payload);
-    store.push(normalizedPayload);
+    if (isEnabled('ds-pushpayload-return')) {
```

Comment:
@igorT I'm not 100% sure I follow.

Currently `store.pushPayload` just looks up the associated serializer and invokes `serializer.pushPayload`. The serializers **uniquely** handle their normalize operations and then callback to the store (`store.push`).

So with all that said, I read your comment as saying, "Can we move this feature check into the store only?" In order to do that, I think you would 1) have to always just return the `store.push` result, or 2) create a new API mechanism for the store to ask the serializer to normalize the payload so the pushPayload` action didn't have to pass through the serializer (which I think is what maybe you were hinting at).

I'm happy to make additional changes or spend additional time continue to improve this feature. Let me know what you'd like me to do.